presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -119,6 +124,23 @@ public void getQueryStateInfos(
         }
     }
 
+    private static boolean matchesWithTimeout(Optional<Pattern> pattern, String input, long timeoutMillis)
```

Comment:
Can the CVE be resolved by doing some kind of sanitation on the regex input instead of running in a thread for a set amount of time?

Reply:
Not entirely. Java already contains some built in checking for evil regex, but there are infinite possible evil regex patterns and they can be very tricky. It is impossible to know if a regex match will hang forever unless actually running the matching algorithm. Cutting off resources at a certain point is the only _100%_ effective way to protect against this kind of vulnerability.